ludwig/data/prompt.py

```diff
@@ -0,0 +1,176 @@
+import json
+import logging
+import os
+import string
+import uuid
+from typing import Any, Dict, Optional, Tuple, TYPE_CHECKING
+
+import pandas as pd
+
+if TYPE_CHECKING:
+    from ludwig.backend.base import Backend
+
+from ludwig.models.retrieval import get_retrieval_model, RetrievalModel
+from ludwig.utils.fs_utils import get_default_cache_location, makedirs, path_exists
+from ludwig.utils.types import Series
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_ZERO_SHOT_PROMPT_TEMPLATE = """SAMPLE INPUT: {sample_input}
+
+USER: Complete the following task: {task}
+
+ASSISTANT:
+"""
+
+
+DEFAULT_FEW_SHOT_PROMPT_TEMPLATE = """Below is relevant context:
+
+CONTEXT: {context}
+
+The context is comprised of labeled samples whose embeddings were similar to that of the sample input. The labels in
```

Comment:
OOC: should context be CONTEXT to match? Or do we lowercase everything pre-tokenization?

Reply:
Made context CONTEXT for clarity.

We do lowercase everything by default, though I think with LLMs we may want to explore the performance deltas between lowercase=True and lowercase=False, since I think it might make a difference particularly in cases like this where it is more like a variable name/identifier than an actual word.